polaris-core/src/main/java/org/apache/polaris/core/persistence/LocalPolarisMetaStoreManagerFactory.java

```diff
@@ -54,7 +54,7 @@ public abstract class LocalPolarisMetaStoreManagerFactory<StoreType>
   final Map<String, Supplier<PolarisMetaStoreSession>> sessionSupplierMap = new HashMap<>();
   protected final PolarisDiagnostics diagServices = new PolarisDefaultDiagServiceImpl();
 
-  protected PolarisStorageIntegrationProvider storageIntegration;
+  @Inject protected PolarisStorageIntegrationProvider storageIntegration;
```

Comment:
This should be removed from here (along with the setter below) because this field is not used in this class.

Instead it should be injected on concrete subtypes that actually need it.